core/src/edu/cornell/gdiac/main/GameController.java

```diff
@@ -324,6 +340,7 @@ public boolean preUpdate(float dt) {
    */
   public void update(float dt) {
     // Process actions in object model
+    InputController input = InputController.getInstance();
     PlayerModel avatar = level.getAvatar();
     avatar.setMovement(InputController.getInstance().getHorizontal() * avatar.getForce());
     avatar.setJumping(InputController.getInstance().didPrimary());
```

Comment:
Probably better to name as `inputController` or similar. We can also get the singleton input controller in the `GameController()` constructor as we also initialize `collisionController` there. Similarly, we can add an final `InputController` field. I think its fine to get the singletons directly, but another option is to inject it into the `GameController(InputController, CollisionController)` etc.

Reply:
Yeah the `getInstance()` method basically ensures we always use the same input controller instance anyway. I do think dependency injection is a bit cleaner here but I think this is fine for the gameplay prototype. We can focus on technical details more for the alpha release.